// @unvt/charites/dist/lib/tileinfo-importer/base-importer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseImporter = void 0;
class BaseImporter {
    constructor(original_urls) {
        const temp_urls = original_urls + '';
        this.urls = temp_urls.split(',');
    }
    async import(style) {
        if (this.urls.length === 0)
            return;
        const responses = await Promise.all(this.urls.map((url) => this.getJSON(url)));
        responses.forEach((res) => {
            Object.keys(res.sources).forEach((sourceName) => {
                style.sources[sourceName] = res.sources[sourceName];
            });
            if (res.layers.length > 0) {
                style.layers = style.layers.concat(res.layers);
            }
        });
        return style;
    }
    getVectorLayers(name, vector_layers) {
        const layers = [];
        if (vector_layers) {
            vector_layers.forEach((layer) => {
                const layerStyle = {
                    id: layer.id,
                    type: 'fill',
                    source: name,
                    'source-layer': layer.id,
                    layout: {},
                    paint: {},
                };
                layers.push(layerStyle);
            });
        }
        return layers;
    }
}
exports.BaseImporter = BaseImporter;
